activesupport/lib/active_support/notifications/fanout.rb

```diff
@@ -182,9 +182,12 @@ def build_event(name, id, payload)
       end
 
       def groups_for(name) # :nodoc:
-        @groups_for.compute_if_absent(name) do
-          listeners_for(name).group_by(&:group_class).transform_values do |s|
-            s.map(&:delegate)
+        groups = @groups_for.compute_if_absent(name) do
+          listeners_for(name).group_by(&:group_class)
```

Comment:
If we expect `silenced?` to change dynamically, this probably needs to be `all_listeners_for`